lib/net/http/header.rb

```diff
@@ -491,7 +491,7 @@ def each_capitalized
   alias canonical_each each_capitalized
 
   def capitalize(name)
-    name.to_s.split(/-/).map {|s| s.capitalize }.join('-')
+    name.to_s.split('-').map {|s| s.capitalize }.join('-')
```

Comment:
Given that the file is `frozen_string_literal: false`: 

```suggestion
    name.to_s.split('-'.freeze).map {|s| s.capitalize }.join('-'.freeze)
```

Probably not a huge deal, but since you are trying to reduce allocations.

Reply:
I was debating hoisting those into constants and freezing them there.

Reply:
A frozen string literal is (very) marginally faster than a constant.

Reply:
>  it may introduce a constant to the public API people may use.

I was thinking about this for another performance improvement, and then remembered it can be `private_constant`d